src/Kunstmaan/NodeBundle/Controller/NodeAdminController.php

```diff
@@ -685,31 +550,14 @@ public function addHomepageAction(Request $request)
         // Check with Acl
         $this->denyAccessUnlessGranted('ROLE_SUPER_ADMIN');
 
-        $type = $this->validatePageType($request);
+        $type = $request->get('type');
+        $title = $request->get('title');
 
-        $newPage = $this->createNewPage($request, $type);
-
-        /* @var Node $nodeNewPage */
-        $nodeNewPage = $this->em->getRepository('KunstmaanNodeBundle:Node')
-            ->createNodeFor($newPage, $this->locale, $this->user);
-        $nodeTranslation = $nodeNewPage->getNodeTranslation(
-            $this->locale,
-            true
-        );
-        $this->em->flush();
+        $nodeTranslation = $this->nodeHelper->createPage($type, $title, $this->locale);
+        $nodeNewPage = $nodeTranslation->getNode();
 
         // Set default permissions
-        $this->container->get('kunstmaan_node.acl_permission_creator_service')
-            ->createPermission($nodeNewPage);
-
-        $nodeVersion = $nodeTranslation->getPublicNodeVersion();
-
-        $this->get('event_dispatcher')->dispatch(
```

Comment:
BC? Order when the event is dispatched has been changed.

Reply:
I have adapted the NodeHelper (doing more BC) so that the event is dispatched at the same momenten as before the change.